test/spec/search.js

```diff
@@ -0,0 +1,376 @@
+      .setValue('#zip-code', '10023')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allIn10023NoRadius);
+  });
+
+  it('should contain all schools within 5 miles of Zip Code 10023', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .setValue('#zip-code', '10023')
+      .setValue('#search-radius', '5')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allIn10023Radius5);
+  });
+
+  // CLARIFICATION: The spec says this should error, but it just returns 0 results?
+  // Returns an error if you define a radius + bogus zip.
+  it('should contain no results for zip code = 1089987987', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .setValue('#zip-code', '1089987987')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, '0');
+  });
+
+  // CLARIFICATION: The spec says this should error, but it just returns all results?
+  it('should contain no results for radius = 0890098', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .setValue('#search-radius', '0890098')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allSchools);
+  });
+
+  // CLARIFICATION: Check include Online-Only - where is that?
+  // CLARIFICATION: Can only select one of the 3 at any given time: State, Region or Zip
+  // How is this limited & what is expected? I just see 0 results.
+
+  /*
+    States
+  */
+  // CLARIFICATION 'FM', 'MP', and 'PW' don't exist
+  it('each state should include the correct count of schools', function*() {
+    for (var index = 0; index < results.allSchoolsInState.length; index++) {
+      var state = results.allSchoolsInState[index];
+      yield browser
+        .url('/')
+        .click('#school-location h1 a')
+        // select by text to ensure the text is the same
+        .selectByVisibleText('.multi-select_item .select-state', state.name);
+        // check selection by value to ensure that value & text are aligned
+      var valueMatch = yield browser
+        .isSelected('.multi-select_item .select-state [value="' + state.code + '"]');
+      yield browser
+        .click('#search-submit');
+
+      var result = yield getResultCount();
+      assert.equal(result, state.count, state.code + ' result count is incorrect. Expected: ' + state.count + ' Actual: ' + result);
+      assert(valueMatch, state.code + ' does not match ' + state.name);
+    }
+  });
+
+  /*
+    Regions 
+  */
+  it('each region should include the correct count of schools', function*() {
+    for (var index = 0; index < results.allSchoolsInRegion.length; index++) {
+      var region = results.allSchoolsInRegion[index];
+      yield browser
```

Comment:
'yield' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).<br>Too many errors. (48% scanned).